src/main/java/core/basesyntax/service/handler/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.handler.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final StorageDao storageDao;
+
+    public BalanceOperationHandler(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void updateStorage(FruitTransaction transaction) {
+        validAmount(transaction);
+        if (!storageDao.isInStorage(transaction.getFruitName())) {
+            storageDao.add(new Fruit(transaction.getFruitName()), transaction.getAmount());
```

Comment:
transaction.getFruitName() can be extracted to a separate variable to avoid code duplication

Reply:
still exists.